### PR TITLE
Implement ab./ablj/ablj - list all the basic blocks ##anal

### PR DESCRIPTION
**Pull request by trufae**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: libr/core/cmd_anal.c [hunk 1]
```diff
@@ -91,14 +91,25 @@ static const char *help_msg_aar[] = {
 static const char *help_msg_ab[] = {
 	"Usage:", "ab", "",
 	"ab", " [addr]", "show basic block information at given address",
+	"ab.", "", "same as: ab $$",
 	"aba", " [addr]", "analyze esil accesses in basic block (see aea?)",
 	"abb", " [length]", "analyze N bytes and extract basic blocks",
 	"abj", " [addr]", "display basic block information in JSON (alias to afbj)",
+	"abl", "[,qj]", "list all basic blocks",
```
Comment (karliss) [inline libr/core/cmd_anal.c:98]: This line uses `,` form for table commands, but help bellow mentions `t`. I am fine with either approach showing or not showing the both foms in helps, but in the case of showing only one of them it should be the same here and in `abl` help.

Comment (trufae) [reply libr/core/cmd_anal.c:98]: There’s a PR from me before the drama that i was trying to get some feedback about this. Basically comma is fine for me as a csv char.. and t is not always available as a subcommand. So inwas trying to discuss about which char to pick here. The inconsistency in this pr about handling comma and t is because the same thing happens in other places in r2.

I have two more PRs for rtable and the big one which would be neat to clarify this syntax decission before going much further on this direction.

Comment (karliss) [reply libr/core/cmd_anal.c:98]: Thanks, for explaining. Missed that discussion.

File: libr/core/cmd_anal.c [hunk 2]
```diff
@@ -2191,6 +2202,231 @@ static ut64 __opaddr(RAnalBlock *b, ut64 addr) {
...
+			}
+			if (block->fcns) {
+				RListIter *iter2;
+				RAnalFunction *fcn;
+				r_list_foreach (block->fcns, iter2, fcn) {
+					r_cons_printf (" .u 0x%" PFMT64x, fcn->addr);
+				}
+			}
+			r_cons_printf (" .s %" PFMT64d "\n", block->size);
+			// r_core_cmdf (core, "pD %"PFMT64d" @ 0x%" PFMT64x, block->size, block->addr);
```
Comment (thestr4ng3r) [inline libr/core/cmd_anal.c:2408]:
```suggestion
```

File: test/db/cmd/cmd_ab [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+.----------------------------------------------------------------------------------------------------------.
+| addr         | size  | traced  | ninstr  | jump         | fail         | fcns          | calls  | xrefs  |
+)----------------------------------------------------------------------------------------------------------(
+| 0x1000011e8  | 41    | 0       | 14      | 0x100001216  | 0x100001211  |  0x1000011e8  |        |        |
```
Comment (thestr4ng3r) [inline test/db/cmd/cmd_ab:20]: Better also add a test where xrefs is not empty to get some coverage there.

Comment (trufae) [reply test/db/cmd/cmd_ab:20]: done

**Review by thestr4ng3r — APPROVED** [review]

**Comment by trufae** [comment]
Will be good to know if that output works for what the people needing full-program cfg. But i think no one will test unless this is merged.

**Comment by trufae** [comment]
Also i think there's some dupe between abl and afb (afb is a subset of abl). and at some point we may be able to render all the output formats using RTable (which can spit r2commands, csv and json). but thats something to do in future prs

File: libr/core/cmd_anal.c [hunk 4]
```diff
@@ -91,14 +91,25 @@ static const char *help_msg_aar[] = {
...
 	"abt[?]", " [addr] [num]", "find num paths from current offset to addr",
 	NULL
 };
 
+static const char *help_msg_abl[] = {
+	"Usage:", "abl", "analyzed basicblocks listing",
+	"abl", "", "list all program-wide basic blocks analyzed",
+	"ablj", "", "in json format",
+	"ablq", "", "in quiet format",
+	"ablt", " [table-query]", "render the list using a table",
```
Comment (trufae) [inline libr/core/cmd_anal.c:109]: abl,
